docs/current_docs/api/custom-types.mdx

```diff
@@ -40,7 +40,7 @@ that returns a custom `Organization` type, itself containing a collection of
 ```typescript file=./snippets/custom-types/typescript/index.ts
 ```
 
-TypeScript has multiple ways to support complex data types. However, the Dagger TypeScript SDK only supports it through fields (decorated with `@func()`) inside classes because `type` and `interface` keywords do not allows decorators.
+TypeScript has multiple ways to support complex data types. Use a `class` when you need methods and privacy, use `type` for plain data objects with only public fields.
```

Comment:
@TomChv should the Python and Go sections also be updated?

Reply:
No, this is specific to TypeScript. For example, in Python you use a class for both, and in Go you also use the same struct for both, so no need to mention.